### PR TITLE
KTOR-5085: Add multiplatform support for XML serialization

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
`ktor-serialization-kotlinx-xml`

**Motivation**
https://youtrack.jetbrains.com/issue/KTOR-5085/Add-multiplatform-support-for-XML

**Solution**
Moved sources and some tests from `jvm` to `common` source set.

Docs change: https://github.com/ktorio/ktor-documentation/pull/513

### CODE REVIEW

File: ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-xml/build.gradle.kts [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+    }
     jvmTest {
         dependencies {
-            api(project(":ktor-client:ktor-client-tests"))
-            api(project(":ktor-server:ktor-server-test-host"))
-            api(project(":ktor-client:ktor-client-plugins:ktor-client-content-negotiation:ktor-client-content-negotiation-tests")) // ktlint-disable max-line-length
-            api(project(":ktor-shared:ktor-serialization:ktor-serialization-kotlinx:ktor-serialization-kotlinx-tests"))
-
-            api(libs.logback.classic)
+            implementation(project(":ktor-client:ktor-client-plugins:ktor-client-content-negotiation:ktor-client-content-negotiation-tests")) // ktlint-disable max-line-length
```
Comment (osipxd) [inline ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-xml/build.gradle.kts:23]: Seems like configuration `implementation` fits better than `api` here, so I replaced it.

File: ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-xml/common/src/io/ktor/serialization/kotlinx/xml/XmlSupport.kt [hunk 2]
```diff
@@ -21,13 +21,16 @@ import nl.adaptivity.xmlutil.serialization.*
...
     repairNamespaces = true
+    // TODO: Remove Suppress and OptIn after update to xmlutil 0.90.2+ or Kotlin 2.0.20+
+    //  https://github.com/pdvrieze/xmlutil/issues/219
+    //  https://youtrack.jetbrains.com/issue/KT-69182/K2-OptIn-on-enum-companion-blocks-enum-constants
+    @Suppress("UnnecessaryOptInAnnotation")
+    @OptIn(XmlUtilInternal::class)
     xmlDeclMode = XmlDeclMode.None
     indentString = ""
     autoPolymorphic = false
-    this.xmlDeclMode
```
Comment (osipxd) [inline ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-xml/common/src/io/ktor/serialization/kotlinx/xml/XmlSupport.kt:30]: Wasn't able to track down the reason why this line is here. If there is the reason, please let me know.

**Comment by e5l** [comment]
@marychatte, could you check?

File: ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-xml/common/src/io/ktor/serialization/kotlinx/xml/XmlSupport.kt (hunk 2 above)
Comment (marychatte) [reply ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-xml/common/src/io/ktor/serialization/kotlinx/xml/XmlSupport.kt:30]: From what I see it's a redundant line, and it does nothing

**Review by marychatte — CHANGES_REQUESTED** [review]
Thanks a lot for the PR!
I just noticed some important things:
1. `gradle build` fails with the next:
```
Execution failed for task ':ktor-shared:ktor-serialization:ktor-serialization-kotlinx:ktor-serialization-kotlinx-xml:transformCommonMainDependenciesMetadata'. 
...
No matching variant of io.github.pdvrieze.xmlutil:serialization:0.90.1 was found. The consumer was configured to find a library for use during 'kotlin-metadata', preferably optimized for non-jvm, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_device_arm64', attribute 'org.jetbrains.kotlin.platform.type' with value 'native' but:
....
- Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_device_arm64'
So I can see from the output that it found watchos_x64, watchos_simulator_arm64  but not watchos_device_arm64
```
So I can see from the output that it found `watchos_x64`, `watchos_simulator_arm64`  but not `watchos_device_arm64`
2. There is also falling test `XmlSerializationTest` for `ios`, `macos` e.t.c.
Because it's a one more whitespace: `Expected <<User id="1" login="vasya"/>>, actual <<User id="1" login="vasya" />>`

**Comment by osipxd** [comment]
@marychatte, thank you for your review!

> 1. `gradle build` fails

Unfortunately, I have only Linux x64 machine at the moment and all apple targets are ignored with message "", so I cannot reproduce this problem.

> 2. There is also falling test `XmlSerializationTest`

Reproduced and fixed. Also, found out that `[js, node]` tests were failing because of [this issue](https://github.com/pdvrieze/xmlutil/issues/83) so I disabled them for now. 

[Here is the correct diff link](https://github.com/ktorio/ktor/compare/c87db9979..b551c29a6) to review the changes.

**Review by marychatte — APPROVED** [review]
Great work!

**Pull request merged**